tests/covscripts/coverage.py

```diff
@@ -0,0 +1,109 @@
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2023 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+
+import json
+import os
+import argparse
+
+COVERAGE_FILE_NAME_TXT = 'develop_percent_coverage.txt'
+COVERAGE_FILE_NAME_JSON = 'coverage.json'
+
+root_dir = os.path.dirname(__file__)
+file_path_curcov_json = os.path.join(root_dir, COVERAGE_FILE_NAME_JSON)
+file_path_prevcov_txt = os.path.join(root_dir, COVERAGE_FILE_NAME_TXT)
+
+
+def check_export(filename):
+  """check if the file was successfully exported"""
+  status = os.path.exists(filename)
+  return status
+
+
+def curcov():
+  """extract current coverage from a json file"""
+  curcov_json = open(file_path_curcov_json)
+  curcov_data = json.load(curcov_json)
+  curcov_percentage = curcov_data['totals']['percent_covered']
+  curcov_percentage_formated = '{:.2f}'.format(curcov_percentage)
+  curcov_percentage_ftdt = float(curcov_percentage_formated)
+  return curcov_percentage_ftdt
+
```

Comment:
```suggestion
def main():
  master_file = sys.argv[1]
  branch_file = sys.argv[2]
  master_pct = curcov(master_file)
  branch_pct = curcov(branch_file)
  if(branch_pct > master_pct)
    sys.exit(0)
  else
    sys.exit(-1)
```

Reply:
I re-implemented the logic in the parse args - allowing it to take file names as an argument. Would that be okay?

Reply:
I think it's better to return an exit code like JD is doing instead of raising an exception if branch_pct < master_pct.